### DISCUSSION TITLE
v0.5.0

### DISCUSSION THREAD

**Discussion by icereed** [icereed/paperless-gpt#45] [Announcements]
# v0.5.0 Release Notes

We are thrilled to announce the release of **Paperless GPT v0.5.0**! This version brings a groundbreaking new feature—**OCR powered by Large Language Models (LLMs)**—along with several enhancements and bug fixes to improve your experience.

---

### OCR via LLMs (Experimental)

- **OCR via LLM** by [@icereed](https://github.com/icereed) in [#29](https://github.com/icereed/paperless-gpt/pull/29)

  Introducing an experimental feature that utilizes LLMs to perform Optical Character Recognition (OCR) on your documents. This advanced OCR method can provide more accurate text extraction, especially in cases where traditional OCR struggles.

  - **Access the Feature**: Enable the feature by specifying `VISION_LLM_PROVIDER` and `VISION_LLM_MODEL` as environment variables.
  - **Processing Time**: Please note that OCR via LLMs may take several minutes to process, depending on the document size and your hardware.
  - **Feedback Welcome**: As this is an early prototype, your feedback is invaluable for further improvements.

---

## 🔧 Improvements

- **Improve Auto Throughput & Logging** by [@JonasHess](https://github.com/JonasHess) in [#40](https://github.com/icereed/paperless-gpt/pull/40)

  - Enhanced the automatic processing pipeline to increase throughput.
  - Improved logging for better monitoring and debugging.

---

## 🐛 Bug Fixes

- **Tag Query Change** by [@ccrlawrence](https://github.com/ccrlawrence) in [#44](https://github.com/icereed/paperless-gpt/pull/44)

  - Improved tag querying to handle edge cases more effectively.
  - Enhanced document retrieval accuracy based on tags.

---

## 📢 Notes

- **Experimental Feature**: The OCR via LLMs is an experimental feature available to all users. Processing times may vary, and results might differ based on document quality and content.
- **Feedback Encouraged**: We invite all users to try out the new OCR feature and share their feedback to help us refine and improve it.
- **Acknowledgments**: A big thank you to all contributors for their hard work and dedication in bringing this release to life.

---

**Upgrade to v0.5.0 today and experience the future of OCR with Paperless GPT!**

## New Contributors
* @ccrlawrence made their first contribution in https://github.com/icereed/paperless-gpt/pull/44

**Full Changelog**: https://github.com/icereed/paperless-gpt/compare/v0.4.0...v0.5.0

<hr /><em>This discussion was created from the release <a href='https://github.com/icereed/paperless-gpt/releases/tag/v0.5.0'>v0.5.0</a>.</em>

**Comment by vibe-code-repo** [comment]
Hey, thanks @all for this release.

Is it possible to mark the documents via tag for a OCR job?

**Reply by icereed** [reply]
Hi @mikekaldig, at the moment not. At the moment, I’m interested in the performance and even correctness of this feature before opening it up as regular feature. But feel free to request a feature how you would like your documents to be consumed for OCR. The next step is indeed to make it nicely usable.
